sphinx/ext/autosummary/__init__.py

```diff
@@ -614,7 +614,7 @@ def get_import_prefixes_from_env(env: BuildEnvironment) -> List[str]:
     return prefixes
 
 
-def import_by_name(name: str, prefixes: List[str] = [None]) -> Tuple[str, Any, Any, str]:
+def import_by_name(name: str, prefixes: List[str] = [None], last_errors = None) -> Tuple[str, Any, Any, str]:
```

Comment:
I feel this change is strange because `last_errors` should be a part of the return value. I understand using argument gives us compatibility. But...

BTW, it should be annotated.

Reply:
Also thought the same, but that would probably break backwards compatibility? Or should I change last_errors to be a retun value?

Reply:
I have two ideas:

1. Add another version of `import_by_name` that also returns list of errors. It will be named as different name.
2. Add a boolean argument like `raise_error` to `import_by_name`. And it raises a custom exception that contains list of errors.

Reply:
Hmm, raise_error would not work that well since we need to collect all errors during the various import methods there are.

Having another version of the import_by_name is IMHO also not such an ideal solution, since then we would need to duplicate code, not only for the `import_by_name` but also for the sub-methods such as `_import_by_name` which is called inside of `import_by_name`